File: onyxsdk-pen/src/androidTest/java/com/onyx/android/sdk/pennative/PenConfig.java
package com.onyx.android.sdk.pennative;

/** Minimal test-only value object matching the installed Notable JNI field contract. */
public final class PenConfig {
    private int type;
    private boolean fastMode;
    private int color = 0xff000000;
    private float width = 3f;
    private float minWidth;
    private int rotateAngle;
    private boolean tiltEnabled;
    private float tiltScale;
    private boolean directionEnabled;
    private float maxTouchPressure = 1f;
    private float dpi = 320f;
    private float displayScaleX = 1f;
    private float displayScaleY = 1f;
    private float scalePrecision = 1f;
    private float brushSpacing = 0.25f;
    private int brushShape;
    private float brushRatio = 5f;
    private float brushAngle;
    private float pressureSensitivity;
    private float velocitySensitivity;
    private float velocityAmplifier;
    private float velocityIgnoreThreshold;
    private float velocityLowerBound;
    private float velocityUpperBound;
    private float smoothLevel = 0.2f;
    private float startPointLimit;
    private float startLengthLimit;
    private float endVelocitySensitivity;
    private float endThinningRate;
    private float ignorePressure;

    public PenConfig setType(int value) { type = value; return this; }
    public PenConfig setFastMode(boolean value) { fastMode = value; return this; }
}

File: onyxsdk-pen/src/main/java/com/onyx/android/sdk/pennative/PenConfig.kt
@file:Suppress(
    // This class intentionally mirrors the mutable, serializable vendor JNI field contract.
    "AvoidVarsExceptWithDelegate",
    "MagicNumber",
    "SerialVersionUIDInSerializableClass",
)

package com.onyx.android.sdk.pennative

import java.io.Serializable

/** Complete configuration contract from `onyxsdk-pennative:1.0.4`. */
open class PenConfig : Serializable {
    var type: Int = 0
    var fastMode: Boolean = false
    var color: Int = 0xff000000.toInt()
    var width: Float = 3f
    var minWidth: Float = 0f
    var rotateAngle: Int = 0
    var tiltEnabled: Boolean = false
    var tiltScale: Float = 0f
    var directionEnabled: Boolean = false
    var maxTouchPressure: Float = 1f
    var dpi: Float = 320f
    var displayScaleX: Float = 1f
    var displayScaleY: Float = 1f
    var scalePrecision: Float = 1f
    var brushSpacing: Float = 0.25f
    var brushShape: Int = NEOPEN_BRUSH_SHAPE_CIRCLE
    var brushRatio: Float = 5f
    var brushAngle: Float = 0f
    var pressureSensitivity: Float = 0f
    var velocitySensitivity: Float = 0f
    var velocityAmplifier: Float = 0f
    var velocityIgnoreThreshold: Float = 0f
    var velocityLowerBound: Float = 0f
    var velocityUpperBound: Float = 0f
    var smoothLevel: Float = 0.2f
    var startPointLimit: Float = 0f
    var startLengthLimit: Float = 0f
    var endVelocitySensitivity: Float = 0f
    var endThinningRate: Float = 0f
    var ignorePressure: Float = 0f

    companion object {
        const val NEOPEN_PEN_TYPE_BRUSH = 1
        const val NEOPEN_PEN_TYPE_FOUNTAIN = 2
        const val NEOPEN_PEN_TYPE_MARKER = 3
        const val NEOPEN_PEN_TYPE_CHARCOAL = 4
        const val NEOPEN_PEN_TYPE_CHARCOAL_V2 = 5
        const val NEOPEN_PEN_TYPE_FOUNTAIN_V2 = 6
        const val NEOPEN_PEN_TYPE_PENCIL = 7
        const val NEOPEN_PEN_TYPE_BALLPOINT = 8
        const val NEOPEN_PEN_TYPE_SQUARE = 9
        const val NEOPEN_PEN_TYPE_BRUSH_SIGN = 10

        const val NEOPEN_BRUSH_SHAPE_CIRCLE = 0
        const val NEOPEN_BRUSH_SHAPE_ELLIPSE = 1
        const val NEOPEN_BRUSH_SHAPE_RECTANGLE = 2
    }
}
